cmd/kubelet/app/server.go

```diff
@@ -243,12 +243,15 @@ func (s *KubeletServer) createAPIServerClient() (*client.Client, error) {
 	if err != nil {
 		return nil, err
 	}
+	if c == nil {
+		return nil, fmt.Errorf("no client returned")
```

Comment:
yeah, probably..   People may wondering why this is necessary because return c, nil is also return nil, nil if c==nil.  It hides the fact that we are doing a nil pointer to nil interface conversion I think.  We may just return nil, nil and comment a little bit?